client/components/tinymce/plugins/wpcom-view/views/embed/index.js

```diff
@@ -46,11 +41,11 @@ export default class EmbedViewManager extends EventEmitter {
 		}
 	}
 
-	addListener( event, listener ) {
+	addListener( event, listener, store ) {
 		super.addListener( event, listener );
-
 		if ( 'change' === event && 1 === this.listeners( event ).length ) {
-			sites.addListener( event, this.sitesListener );
+			this.store = store;
+			this.unsubscribeStore = store.subscribe( this.sitesListener );
```

Comment:
Hi @gwwar,

You are right it will fire the event every time the store changes, but then our listener will just execute getSelectedSiteId and if it did not change the listener does nothing, so the performance hit is just executing getSelectedSiteId and doing an if.
Unfortunately, in redux there is no way to subscribe to just some parts of the state, each listener always subscribes to everything and is responsible for verifying if the change is relevant to it or not and we are doing that, so we are not far from the redux way of doing things. But yes I also don't like the idea of subscribing it there.

Yes, it may be possible to add a selectedSiteId field to EmbedsStore and then subscribe to it on the EventEmitter, but then each time the selectedSiteId changes the event will be sent to the store to set the selectedSiteId even when we are not the editor right? And I think if we add EmbedsStore to lib/middleware we will be loading EmbedsStore code in every part of calypso that uses lib/middleware (probably everywhere). Or maybe I did understood well the way to use lib/middleware, I will  need to check in more detail.